monkey/monkey_island/cc/services/authentication_service/flask_resources/agent_otp.py

```diff
@@ -12,12 +17,33 @@ class AgentOTP(AbstractResource):
     """
 
     urls = ["/api/agent-otp"]
+    lock = Lock()
+    limiter = None
+
+    def __init__(self, limiter: Limiter):
+        # Since flask generates a new instance of this class for each request,
+        # we need to ensure that a single instance of the limiter is used. Hence
+        # the class variable.
+        #
+        # TODO: The limit is currently applied per IP address. We will want to change
+        # it to per-user once we require authentication for this endpoint.
+        with AgentOTP.lock:
+            if AgentOTP.limiter is None:
+                AgentOTP.limiter = limiter.limit(
+                    "10/second", key_func=get_remote_address, per_method=True
```

Comment:
It might be good to put the number from the limit into a constant. This will let BB tests import and uses it.

50 might be a better limit.